### PR TITLE
Bugfix: Constraint Temporary Vector Usage

### PR DESCRIPTION
**Pull request by gardner48**
Fix a potential bug when constraint handling is enabled and `GetEstLocalErrors`is called in ARKODE

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ internal solution vector.  Users who wish to revert to interpolation may call a
 routine `CVodeSetInterpolateStopTime`, `ARKStepSetInterpolateStopTime`,
 `ERKStepSetInterpolateStopTime`, or `MRIStepSetInterpolateStopTime`.
 
+A potential bug was fixed when using inequality constraint handling and
+calling `ARKStepGetEstLocalErrors` or `ERKStepGetEstLocalErrors` after a failed
+step in which an inequality constraint violation occurred. In this case, the
+values returned by `ARKStepGetEstLocalErrors` or `ERKStepGetEstLocalErrors` may
+be invalid.
```
Comment (drreynolds) [inline CHANGELOG.md:24]:
```suggestion
have been invalid.
```

File: doc/arkode/guide/source/Introduction.rst [hunk 2]
```diff
@@ -137,6 +137,12 @@ internal solution vector.  Users who wish to revert to interpolation may call a
 routine :c:func:`ARKStepSetInterpolateStopTime`,
 :c:func:`ERKStepSetInterpolateStopTime`, or :c:func:`MRIStepSetInterpolateStopTime`.
 
+A potential bug was fixed when using inequality constraint handling and
+calling :c:func:`ARKStepGetEstLocalErrors` or :c:func:`ERKStepGetEstLocalErrors`
+after a failed step in which an inequality constraint violation occurred. In
+this case, the values returned by :c:func:`ARKStepGetEstLocalErrors` or
+:c:func:`ERKStepGetEstLocalErrors` may be invalid.
```
Comment (drreynolds) [inline doc/arkode/guide/source/Introduction.rst:144]:
```suggestion
:c:func:`ERKStepGetEstLocalErrors` may have been invalid.
```

File: examples/arkode/CXX_serial/ark_heat2D.cpp
Comment (drreynolds) [inline examples/arkode/CXX_serial/ark_heat2D.cpp:1]: I'm guessing that the changes in this file are unrelated to this PR.  What did these fix?

**Review by drreynolds — APPROVED** [review]
A couple of wording suggestions and a question, but the PR itself looks good to go.

File: examples/arkode/CXX_serial/ark_heat2D.cpp
Comment (gardner48) [reply examples/arkode/CXX_serial/ark_heat2D.cpp:1]: This fixes an uninitialized value warning that comes up with GCC 12
